sw/device/lib/dif/dif_plic.h

```diff
@@ -0,0 +1,248 @@
+  kDifPlicGpio27 = 28,                  /**< GPIO pin 27 */
+  kDifPlicGpio28 = 29,                  /**< GPIO pin 28 */
+  kDifPlicGpio29 = 30,                  /**< GPIO pin 29 */
+  kDifPlicGpio30 = 31,                  /**< GPIO pin 30 */
+  kDifPlicGpio31 = 32,                  /**< GPIO pin 31 */
+  kDifPlicUartTxWatermark = 33,         /**< UART TX FIFO watermark */
+  kDifPlicUartRxWatermark = 34,         /**< UART RX FIFO watermark */
+  kDifPlicUartTxOverflow = 35,          /**< UART TX FIFO overflow */
+  kDifPlicUartRxOverflow = 36,          /**< UART RX FIFO overflow */
+  kDifPlicUartRxFrameErr = 37,          /**< UART RX frame error */
+  kDifPlicUartRxBreakErr = 38,          /**< UART RX break error */
+  kDifPlicUartRxTimeout = 39,           /**< UART RX timeout */
+  kDifPlicUartRxParityErr = 40,         /**< UART RX parity error */
+  kDifPlicSpiDeviceRxF = 41,            /**< SPI RX SRAM FIFO full */
+  kDifPlicSpiDeviceRxLvl = 42,          /**< SPI RX SRAM FIFO above the level */
+  kDifPlicSpiDeviceTxLvl = 43,          /**< SPI TX SRAM FIFO under the level */
+  kDifPlicSpiDeviceRxErr = 44,          /**< SPI MOSI in FwMode has error */
+  kDifPlicSpiDeviceRxOverflow = 45,     /**< SPI RX Async FIFO overflow */
+  kDifPlicSpiDeviceTxUnderflow = 46,    /**< SPI TX Async FIFO underflow */
+  kDifPlicFlashCtrlProgEmpty = 47,      /**< Flash prog FIFO empty */
+  kDifPlicFlashCtrlProgLvl = 48,        /**< Flash prog FIFO drained to lvl */
+  kDifPlicFlashCtrlRdFull = 49,         /**< Flash read FIFO full */
+  kDifPlicFlashCtrlRdLvl = 50,          /**< Flash read FIFO filled to lvl */
+  kDifPlicFlashCtrlOpDone = 51,         /**< Flash operation complete */
+  kDifPlicFlashCtrlOpError = 52,        /**< Flash operation failed with err */
+  kDifPlicHmacDone = 53,                /**< HMAC done */
+  kDifPlicHmacFifoFull = 54,            /**< HMAC FIFO full */
+  kDifPlicHmacErr = 55,                 /**< HMAC error */
+  kDifPlicAlertHandlerClassA = 56,      /**< Alert Handler class A alert */
+  kDifPlicAlertHandlerClassB = 57,      /**< Alert Handler class B alert */
+  kDifPlicAlertHandlerClassC = 58,      /**< Alert Handler class C alert */
+  kDifPlicAlertHandlerClassD = 59,      /**< Alert Handler class D alert */
+  kDifPlicNmiGenEsc0 = 60,              /**< NMI Gen escalation interrupt 0 */
+  kDifPlicNmiGenEsc1 = 61,              /**< NMI Gen escalation interrupt 1 */
+  kDifPlicNmiGenEsc2 = 62,              /**< NMI Gen escalation interrupt 2 */
+  kDifPlicNmiGenEsc3 = 63,              /**< NMI Gen escalation interrupt 3 */
+  kDifPlicLastIrq = kDifPlicNmiGenEsc3, /**< The last valid IRQ ID */
+} dif_plic_irq_src_id_t;
+
+/**
+ * PLIC IRQ source peripheral enumeration
+ *
+ * Enumeration used to determine which peripheral asserted the corresponding
+ * IRQ source.
+ */
+typedef enum dif_plic_irq_peripheral {
+  kDifPlicPeripheralNone = 0, /**< none */
+  kDifPlicGpio,               /**< GPIO */
```

Comment:
I think these names should have more type specificity.  For instance `kDifPlicGpio` and `kDifPlicGpio0` are different types, but neither of them are indicative of their intended use.

Reply:
I have now changes the corresponding type names and constant names.